cmd/zqd/listen/command.go

```diff
@@ -85,12 +89,10 @@ func (c *Command) Run(args []string) error {
 	if c.pprof {
 		h = pprofHandlers(h)
 	}
-	ln, err := net.Listen("tcp", c.listenAddr)
-	if err != nil {
-		return err
-	}
-	c.logger.Info("Listening", zap.Stringer("addr", ln.Addr()))
-	return http.Serve(ln, h)
+	ctx := c.catcher.Catch(context.Background())
```

Comment:
I think it would be clearer to ditch catcher and do this:
```suggestion
	ctx, cancel := context.WithCancel(context.Background())
	ch := make(chan os.Signal, 1)
	signal.Notify(ch, os.Interrupt)
	go func() {
		sig := <- ch
		c.logger.Info("Signal received", zap.Stringer("signal", sig))
		cancel()
	}()
```

Reply:
I agree with @nwt : on the whole, I think its simpler to just handle the signals & cancellation in these few lines of code directly where its needed.

Reply:
ok

Reply:
For the record, I still think `catcher` looks prettier than this.